tools/plugin/alsaplug/tplg.c

```diff
@@ -1456,7 +1456,12 @@ void plug_free_topology(snd_sof_plug_t *plug)
 
 	list_for_item_safe(item, _item, &plug->widget_list) {
 		struct tplg_comp_info *comp_info = container_of(item, struct tplg_comp_info, item);
+		struct sof_ipc4_available_audio_format *available_fmts = &comp_info->available_fmt;
 
+		if (available_fmts->output_pin_fmts)
+			free(available_fmts->output_pin_fmts);
+		if (available_fmts->input_pin_fmts)
+			free(available_fmts->input_pin_fmts);
```

Comment:
Isn't this if-check redundannt, NULL is ok to free

Reply:
ack fixed now

Reply:
@ranj063 the fix went into the wrong patch